mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/events/ElasticSearchListeningMessageSearchIndex.java

```diff
@@ -62,16 +63,16 @@
     private static final Logger LOGGER = LoggerFactory.getLogger(ElasticSearchListeningMessageSearchIndex.class);
     private static final String ID_SEPARATOR = ":";
 
-    private final ElasticSearchIndexer indexer;
+    private final Indexer indexer;
     private final ElasticSearchSearcher searcher;
     private final MessageToElasticSearchJson messageToElasticSearchJson;
 
     @Inject
     public ElasticSearchListeningMessageSearchIndex(MessageMapperFactory factory,
-            @Named(ElasticSearchConstants.MAILBOX_INDEX) ElasticSearchIndexer indexer,
+            @Named(ElasticSearchConstants.MAILBOX_INDEX) IndexerSupplier indexer,
```

Comment:
the Supplier add nothing to the design here.

Reply:
I do disagree. See below comment.